mergin/test/test_client.py

```diff
@@ -1945,3 +1946,26 @@ def test_clean_diff_files(mc):
     diff_files = glob.glob("*-diff-*", root_dir=os.path.split(mp.fpath_meta("inserted_1_A.gpkg"))[0])
 
     assert diff_files == []
+
+
+def test_download_failure(mc):
+    test_project = "test_download_failure"
+    project = API_USER + "/" + test_project
+    project_dir = os.path.join(TMP_DIR, test_project)
+    download_dir = os.path.join(TMP_DIR, "download", test_project)
+
+    cleanup(mc, project, [project_dir, download_dir])
+    # prepare local project
+    shutil.copytree(TEST_DATA_DIR, project_dir)
+
+    # create remote project
+    mc.create_project_and_push(test_project, directory=project_dir)
+
+    # download project async
+    with pytest.raises(IsADirectoryError):
+        job = download_project_async(mc, project, download_dir)
+        os.makedirs(os.path.join(download_dir, "base.gpkg.0"))
```

Comment:
No it would not. We can simulate failure by creating a dummy `base.gpkg` file and making it immutable with `chattr`. But, in this case exception will happen at the finalize step and as far as I can see we don't remove downloaded project at that stage (maybe we should?) so log remains on disk and we don't need to copy it anywhere